### PR TITLE
Develop

### LINKED ISSUE #335: [TARA ERC-20 INFO] Implement required endpoints for new token details website
We need 3 additional routes, all of these reflecting the amounts on the Taraxa Mainnet(ETH Mainnet):

- staking
- token(already there)
- validators

**Staking endpoints**

Returns:

- Total stake (everything in the staking contract) 
- Total delegation (we have this number on the community site)
- Average validator commission % (average commission rate weighted by their delegation)

**Token endpoints**

Returns:

- Total locked (however much is still in the deployer address: 0x723304D1357A2334FCf902Aa3d232f5139080A1b)
- Total circulation (can be calculated: total supply - total locked - total staked) BUT CoinGecko needs an API so we need to give them one 

**Validators endpoints**

Returns:

- Number of block-producing mainnet validators
- Number of block-producing testnet validators 
- Total cumulative commission earned by mainnet validators (hope the community site has this)
- Total cumulative rewards earned by testnet validators (hope the community site has this)
- Total # github commits in the last 30-days (unsure if this is available without an expensive Github upgrade… I’m hoping we can all repos private + public in the taraxa project organization, right now we’re hand-counting them once a month for our update, be nice to get an automated #) https://stackoverflow.com/questions/27931139/how-to-use-github-v3-api-to-get-commit-count-for-a-repo

### PR DESCRIPTION
**Pull request by VargaElod23**

### CODE REVIEW

File: src/github/github.service.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  async getRepoNames() {
+    let returnData;
+    let endCursor;
+    const repoNames = [];
+    let isDone = false;
+    while (!isDone) {
+      try {
+        const token = this.configService.get<string>('githubAccessToken');
+        this.logger.log(token);
```
Comment (rattrap) [inline src/github/github.service.ts]: Let's remove this

File: src/node/node.service.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  async noActiveValidators(testnet?: boolean) {
+    let explorerRoot;
+    if (testnet) {
+      explorerRoot = this.configService.get<string>('testnetExplorerRoot');
+    } else {
+      explorerRoot = this.configService.get<string>('explorerRoot');
+    }
+    const explorerApi = `${explorerRoot}/api/nodes?week=${getWeek(
+      1,
+    )}&year=${new Date().getUTCFullYear()}`;
```
Comment (rattrap) [inline src/node/node.service.ts]: The year/week thing is too complicated. The API already has these default values. Just call it with `/api/nodes?limit=1000`

File: src/token/token.controller.ts [hunk 3]
```diff
@@ -21,14 +23,65 @@ export class TokenController {
   async getSymbol() {
     return await this.tokenService.getSymbol();
   }
+  /**
+   * Returns the current TARA price
+   * @returns price as float
+   */
+  @Get('price')
+  @CacheTTL(36000)
```
Comment (rattrap) [inline src/token/token.controller.ts]: We need this to update faster. Let's just have 30 secs here

File: src/utils/date.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const getWeek = function (dowOffset: number) {
```
Comment (rattrap) [inline src/utils/date.ts]: Let's remove this whole thing 'cause it looks too complicated

File: src/utils/types.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+export interface ValidatorBasicInfo {
```
Comment (rattrap) [inline src/utils/types.ts]: Do we use this anywhere ?

**Review by rattrap — APPROVED** [review]

File: src/utils/types.ts (hunk 5 above)
Comment (VargaElod23) [reply src/utils/types.ts]: No, only the following one

**Pull request merged**